ipamd/ipamd.go

```diff
@@ -1046,10 +1059,27 @@ func getWarmIPTarget() int {
 	return noWarmIPTarget
 }
 
-// ipTargetState determines the number of IPs `short` or `over` our WARM_IP_TARGET
+func getMinimumIPTarget() int {
+	inputStr, found := os.LookupEnv(envMinimumIPTarget)
+
+	if !found {
+		return noMinimumIPTarget
+	}
+
+	if input, err := strconv.Atoi(inputStr); err == nil {
+		if input >= 0 {
```

Comment:
We do know the maximum possible value here as well, `c.maxENI  * c.maxIPsPerENI`. It's probably best to check that as well, since on line 1099 you do `over = min(over, total-c.minimumIPTarget)`, and that would return a negative value if someone set `MINIMUM_IP_TARGET=1000`.

Reply:
We don't know `c.maxENI` / `c.maxIPsPerENI` yet because that is calculated as part of `nodeInit()`, which is called after this method is called during `New` and `nodeInit()` depends on `c.ipTargetState()` having these values already. I don't see anything obvious preventing additional refactoring to make these specific values available earlier, but could you please walk me through a bit of the motivation? It seems like this behavior is now the same as it was with a very large `WARM_IP_TARGET`.

I've updated line 1099 to prevent returning a negative number. I had assumed that calling code would treat negative numbers the same as 0, but it looks like that's not the case. Great catch!

Reply:
@asheldon Ah, you're right. We need the instance type first... Thanks for adding the check.